airsonic-main/src/main/java/org/airsonic/player/service/SecurityService.java

```diff
@@ -55,11 +66,37 @@
 
     @Autowired
     private UserDao userDao;
+
+    @Autowired
+    private SonosLinkDao sonosLinkDao;
+
     @Autowired
     private SettingsService settingsService;
+
     @Autowired
     private Ehcache userCache;
 
+    @Autowired
+    private Ehcache sonosLinkcodeCache;
+
+    @Autowired
+    private JWTSecurityService jwtSecurityService;
+
+    /*
+        TODO This is initialized in GlobalSecurityConfig.
+
+        Something wrong here, some circular ref, maybe rebuild responsibilities...
+
+        We are :
+           - RESTRequestParameterProcessingFilter need SecurityService cause is UserDetailsService.
+           - But SecurityService need AuthenticationManager they create in GlobalSecurityConfig
+           - But GlobalSecurityConfig create RESTRequestParameterProcessingFilter
+
+           So GlobalSecurityConfig -> RESTRequestParameterProcessingFilter -> SecurityService -> AuthenticationManager (but create in GlobalSecurityConfig)
+     */
+    private AuthenticationManager authenticationManager;
```

Comment:
this bit is weird, segregation of responsibilities or maybe a clue stuff shouldn't be organized this way (security service shouldn't have all this?)